### PR TITLE
Updating integration with DataFrames

### PR DESCRIPTION
**Pull request by racinmat**
Updating documentation describing DataFrames integration so it works with newest versions.
Adding DataFrames as test dependency, adding tests for the integration.
I know adding more dependnecies is a bit opinionated, but I think it should be tested if it's covered in documentation.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -11,16 +11,17 @@ SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 
 [compat]
-LearnBase = "0.2, 0.3"
+LearnBase = "0.2, 0.3, 0.4"
```
Comment (johnnychen94) [inline Project.toml:14]: MLLabelUtils isn't yet compatible with LearnBase `v"0.4"`, which is a slightly breaking version. Check the CI logs you'll find it's `v"0.3"` that's used.

```suggestion
LearnBase = "0.2, 0.3"
```

Comment (racinmat) [reply Project.toml]: Oh, thanks, my bad. I saw that tests in the PR https://github.com/JuliaML/MLDataPattern.jl/pull/45 are passing, but did not notice that it's actually not used. Will revert this.

File: docs/documentation/targets.rst [hunk 2]
```diff
@@ -248,7 +248,7 @@ column ``:y`` contains the targets.
    │ 4   │ 0.582912  │ 0.509047 │ a │
    │ 5   │ 0.407289  │ 0.113006 │ b │
 
-   julia> targets(row->row[1,:y], df)
+   julia> targets(row->row[:y], df)
```
Comment (oxinabox) [inline docs/documentation/targets.rst:251]:
```suggestion
   julia> targets(row->row.y, df)
```

File: docs/documentation/targets.rst [hunk 3]
```diff
@@ -421,7 +421,7 @@ we will assume that the column ``:y`` contains the targets.
    │ 4   │ 0.582912  │ 0.509047 │ a │
    │ 5   │ 0.407289  │ 0.113006 │ b │
 
-   julia> iter = eachtarget(row->row[1,:y], df)
+   julia> iter = eachtarget(row->row[:y], df)
```
Comment (oxinabox) [inline docs/documentation/targets.rst:424]:
```suggestion
   julia> iter = eachtarget(row->row.y, df)
```

File: docs/documentation/targets.rst [hunk 4]
```diff
@@ -590,7 +590,7 @@ be done generically by specifying a target-extraction-function.
    │ 4   │ 0.522172 │ 0.812814  │ a │
    │ 5   │ 0.505208 │ 0.245457  │ b │
 
-   julia> targets(row->row[1,:y], df)
+   julia> targets(row->row[:y], df)
```
Comment (oxinabox) [inline docs/documentation/targets.rst:593]:
```suggestion
   julia> targets(row->row.y, df)
```

File: test/tst_dataframes.jl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @testset "targets" begin
+        y = [:a,:a,:b,:a,:b]
+        df = DataFrame(x1 = rand(5), x2 = rand(5), y = y)
+        @test y == targets(row->row[:y], df)
+    end
+
+    @testset "eachtarget" begin
+        y = [:a,:a,:b,:a,:b]
+        df = DataFrame(x1 = rand(5), x2 = rand(5), y = y)
+        iter = eachtarget(row->row[:y], df)
```
Comment (oxinabox) [inline test/tst_dataframes.jl:14]:
```suggestion
        iter = eachtarget(row->row.y, df)
```

File: test/tst_dataframes.jl [hunk 6]
```diff
@@ -0,0 +1,31 @@
+@testset "DataFrame integration" begin
+    LearnBase.getobs(df::DataFrame, idx) = df[idx,:]
+    LearnBase.nobs(df::DataFrame) = nrow(df)
+
+    @testset "targets" begin
+        y = [:a,:a,:b,:a,:b]
+        df = DataFrame(x1 = rand(5), x2 = rand(5), y = y)
+        @test y == targets(row->row[:y], df)
```
Comment (oxinabox) [inline test/tst_dataframes.jl:8]:
```suggestion
        @test y == targets(row->row.y, df)
```

File: Project.toml [hunk 7]
```diff
@@ -17,10 +17,11 @@ StatsBase = "0.24, 0.25, 0.26, 0.27, 0.28, 0.29, 0.30, 0.31, 0.32, 0.33"
 julia = "0.7, 1"
 
 [extras]
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
```
Comment (oxinabox) [inline Project.toml:20]: need to specify Compat for DataFrames
else tests will break randomly at some point.

Comment (johnnychen94) [reply Project.toml:20]: Just add it in the `[compat]` as if it's normal deps would work.

Comment (racinmat) [reply Project.toml:21]: Added

**Comment by johnnychen94** [comment]
Thanks again for working on this

**Pull request merged**